### DISCUSSION TITLE
Qucs-S 1.1.0

### DISCUSSION THREAD

**Discussion by ra3xdh** [ra3xdh/qucs_s#278] [Announcements]
## New features

* Qucs-S supports digital simulation mode using IcarusVerilog or FreeHDL backends #97
* Added analog model for digital source #265
* Qucs-S runs DC bias simulation mode if schematic contains DC simulation only #271
* Selection rectangle update #276

## Deprecated features

* XSPICE CodeModel synthesizer has been deprecated since v1.1.0. The precompiled CM libraries
  are still supported using *spiceinit* virtual device. See #204 

## Known issues of digital simulation

* Digital buses are not supported;
* Digital simulation represents its state as of Qucs-0.0.20-RC1; No new features added

### Source code

Download the source code file `qucs-s-1.1.0.tar.gz` and build it with `cmake` to try new Qucs-S. 

### Linux

* AppImage for all Linux distributions;  download file `Qucs-S-v1.1.0.x86_64.AppImage`
*  DEB and RPM packages for Debian, Ubunut, Fedora, CentOS, openSUSE generated with OBS. Find the repository for your distribution here: https://software.opensuse.org/download.html?project=home%3Ara3xdh&package=qucs-s

### Windows

Windows portable package could be downloaded as zipped distribution. Unpack and launch `qucs-s.exe` from `bin` subdirectory. Ngspice requires a separate installation. There is two windows packages available:
* `qucs_s_win64_1_1_0.zip` 64-bit (x86_64) portable Windows version;
* `qucs_s_win32_1_1_0.zip` 32-bit (i686) portable Windows version;

**Comment by csrabak** [comment]
Before I open, a perhaps unecessary, issue, I would like to know what should be expected of properness of the examples available at
```share\qucs-s\examples\ngspice\``` because when I try the active bandpass filter example ```active_bp.sch``` even after running (via F8) the simulation, the plot ordinate shows a "ngspice/ac.v(out) invalid". Whereas the log file (via F5) only shows "ngspice starting" and the DC voltages in blue indicate the simulation has run.

Furthermore, to my confusion, if I select, say, ac.v(out) available in the properties of the plot, I get a family of curves and the name of data becomes "BJT-swp:ac.v(out)" which I can surmise has been a command for a parameter sweep elsewhere, but it is not evident in the diagram!

Looking at the generated netlist also does not enlighten the issue.

I did not make an exhaustive analysis of every example, so let's take this as sample.

I also understand there is a burden in updating the docs, even in Readthedocs while the project is progressing.

That's why I made this question at the start of this post.

Also, please accept my kudos for more this milestone!

**Comment by ra3xdh** [comment]
> even after running (via F8)

I cannot reproduce the reported problem with the simulation. This schematic is not broken. Everything works as expected on my machine. The F8 mode runs only DC bias simulation but not the full simulation. And the plot will remain empty. Did you try to launch "Simulation->Simulate" or F2?

The Ngspice examples were revised recently. Incompatible example are moved away from this subdirectory. See #162 

![image](https://github.com/ra3xdh/qucs_s/assets/4920080/4364b8ee-3d86-4b35-bbef-39170f40783b)

**Comment by csrabak** [comment]
My present version of qucs-s is:
![image](https://github.com/ra3xdh/qucs_s/assets/10200568/a32cba4c-5513-441a-aaf2-3dc45bfc53f5)
My OS is:
Edição	Windows 10 Home Single Language
Versão	22H2
Instalado em	‎08/‎11/‎2020
Compilação do SO	19045.2965
Experiência	Windows Feature Experience Pack 1000.19041.1000.0
My NGSpice is:
```******
** ngspice-40 : Circuit level simulation program
** The U. C. Berkeley CAD Group
** Copyright 1985-1994, Regents of the University of California.
** Copyright 2001-2023, The ngspice team.
** Please get your ngspice manual from https://ngspice.sourceforge.io/docs.html
** Please file your bug-reports at https://ngspice.sourceforge.io/bugrep.html
** Creation Date: Mar 27 2023   15:24:42
******```

The ```active_bp.sch``` is pristine from ver 1.1.0 installation:
```
C:\Arquivos de Programas\qucs_s_win64\share\qucs-s\examples\ngspice>dir active_bp.sch

30/10/2022  08:54             2.006 active_bp.sch```

When I pressed F8 I got:

![image](https://github.com/ra3xdh/qucs_s/assets/10200568/1cc86728-5151-4477-9fdc-f8b92c740f47)

Pressing F2:

![image](https://github.com/ra3xdh/qucs_s/assets/10200568/648f22ef-11cd-4b39-93ba-cbb76c5028b7)

If I open the properties of the diagram I get these options:

![image](https://github.com/ra3xdh/qucs_s/assets/10200568/0aaed878-78fd-460d-9d7f-22070356b3ff)

Attempting to be intelligent, I checked the dropbox to see if a set for ngspice data was available:

![image](https://github.com/ra3xdh/qucs_s/assets/10200568/7f3a1be2-5f6f-4e02-9f45-fd10f932b278)

My expectation was that one of the datasets where "active_bp" but I got those puzzings "BJT-swp" and "flip-flops_truth_tables", which see, are names of simulations I never have done with qucs-s (and no other qucs related simulation whatsover).

The only non standard configuration for qucs-s is that I change the default "Qucs home" from ```~/.qucs``` to ```~/.qucs`-s``` in order to avoid clashes with legacy projects done with Qucs 0.0.19 (or Qucs Studio, which I also moved the home for the projects of that version).

Since it works in your machine I surmise some detail on my particular installation is a probable cause...

**Comment by ra3xdh** [comment]
It looks very strange. The Ngspice and Qucs-S fail to create a datset, but the schematic contains no error.

I still cannot reproduce your problem. Everything works as expected on my system. Can you simulate any other circuit? Did you try to create a simple circuit from scratch in your home directory and simulate it? Do you have write permissions on the example directory? Are you running simulation from Windows user or administrator? Also I suspect some software like anti-virus may prevent Ngspice from writing a dataset.  

> "BJT-swp" and "flip-flops_truth_tables", which see, are names of simulations I never have done with qucs-s

The datasets cannot come from nowhere. Look for `*.dat.ngspice` files.

**Comment by ra3xdh** [comment]
Also do you see the netlist is written after pressing Simulate->Show Last netlist?

**Reply by csrabak** [reply]
Yes I see it.
```
* Qucs 1.1.0 C:/Arquivos de Programas/qucs_s_win64/share/qucs-s/examples/ngspice/active_bp.sch
.INCLUDE "C:/Arquivos de Programas/qucs_s_win64/share/qucs-s/xspice_cmlib/include/ngspice_mathfunc.inc"
* Qucs 1.1.0  C:/Arquivos de Programas/qucs_s_win64/share/qucs-s/examples/ngspice/active_bp.sch
B_OP1 out 0 V = 1E6*V(_net0,0)*u(15-1E6*V(_net0,0))*u(1E6*V(_net0,0)-(-15))+15*u(1E6*V(_net0,0)-15)+(-15)*u((-15)-1E6*V(_net0,0))
R3 in _net1  390
C1 _net1 _net0  47N 
C2 _net1 out  47N 
R1 _net0 out  3.9K
V1 in 0 DC 0 SIN(0 1 1G 0 0 0) AC 1 ACPHASE 0
R2 0 _net1  5.6K
.control
echo "" > spice4qucs.cir.noise
echo "" > spice4qucs.cir.pz
ac dec 67 0.1k 100k 
write active_bp_ac.txt v(in) v(out) 
destroy all
reset

exit
.endc
.END
```
As I see the "destroy all" I didn't bother to search for any other file...

**Comment by ra3xdh** [comment]
Also try to copy schematic from `Program Files` to other place. Windows may prevent modification of this directory.

**Reply by csrabak** [reply]
OK, I'll try this, but for debugging purposes and in 'my defense' since I see the log.txt correctly written to the home directory I pointed to, I was surmising the necessary files would be written there.

If this is an aspect to be considered, please accept my suggestion to have some ways the GUI to alert the user.

A note in the README (and perhaps in the docs) could also be a worthwhile addition.

**Comment by ra3xdh** [comment]
Dataset files should be created in the same directory where schematic file is located. And therefore you need to save schematic in the place where you have write permission. Qucs-S show a warning when schematic is opened in Read-Only mode. See #224 This feature works for the latest `1.1.0` and is not broken. It works fine when examples are opened on Linux from `/usr/share/qucs-s/examples`. Bit it seems it cannot detect the case when the directory is protected by UAC or similar defence mechanics.

**Comment by ra3xdh** [comment]
Windows filesystem defence may work differently and may require another approach. I will look into this.

**Comment by ra3xdh** [comment]
Yes, I can reproduce it now using Windows 10 installation. The issue takes place only if the examples are installed in the protected directory. But the bug doesn't appear if Qucs-S is installed in disk root like `C:\qucs_win64`. The `Program Files` has write protection by UAC which is not detected by Qucs application.

**Comment by ra3xdh** [comment]
I never used the `Program Files` for the test installation on Windows before. That's why this problem remain unnoticed for a long time.

**Reply by csrabak** [reply]
OK, I'm also learning a lot having to work in this SO after some years...
In Linux things happen slightly different as you already reported.
I'll copy the whole set of examples to a folder under my home dir, and keep learning.

Thanks for the attention.

73

**Comment by ra3xdh** [comment]
See #282

**Comment by ra3xdh** [comment]
I have added a warning message if dataset cannot be created. See #282. This improvement will be available since the next release.
